src/rules/no-important.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @fileoverview Rule to disallow !important flags.
+ */
+
+//-----------------------------------------------------------------------------
+// Imports
+//-----------------------------------------------------------------------------
+
+import { findOffsets } from "../util.js";
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @import { CSSRuleDefinition } from "../types.js"
+ * @typedef {"unexpectedImportant"} NoImportantMessageIds
+ * @typedef {CSSRuleDefinition<{ RuleOptions: [], MessageIds: NoImportantMessageIds }>} NoImportantRuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoImportantRuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			description: "Disallow !important flags",
+			recommended: true,
+			url: "https://github.com/eslint/css/blob/main/docs/rules/no-important.md",
+		},
+
+		messages: {
+			unexpectedImportant: "Unexpected !important flag found.",
+		},
+	},
+
+	create(context) {
+		return {
+			Declaration(node) {
+				if (node.important) {
+					const declarationText = context.sourceCode.getText(node);
+					const importantPattern = /!(\s|\/\*.*?\*\/)*important/iu;
```

Comment:
Because this regular expression never changes, we can move it outside of this method to avoid re-parsing every time `Declaration` is called.